cmd/controller-manager/app/controller-manager.go

```diff
@@ -117,6 +119,25 @@ func Run(opts *options.Options) error {
 		klog.Fatalf("Invalid Feature Gate: %v", err)
 	}
 
+	// validate if a federation namespace is configured
+	if len(opts.Config.FederationNamespace) == 0 {
+		klog.Fatalf("The federation namespace must be specified via --federation-namespace")
+	}
+
+	if len(masterURL) == 0 && len(kubeconfig) == 0 && len(opts.Config.FederationNamespace) == 0 {
+		// for in-cluster deployment set the namespace associated
+		// with the service account token
+		data, err := ioutil.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/namespace")
+		if err != nil {
+			klog.Fatalf("An error occurred while attempting to discover the federation namespace from the service account: %v", err)
+		} else {
```

Comment:
This else block is unnecessary since `klog.Fatalf` will exit the process.

Reply:
Done!